subsys/net/lib/lwm2m/lwm2m_engine.c

```diff
@@ -2385,17 +2451,42 @@ static int handle_request(struct zoap_packet *request,
 		}
 
 		/* 0 will be returned if it's the last block */
-		block_offset = zoap_next_block(in.in_zpkt, &block_ctx->ctx);
+		block_offset = coap_next_block(in.in_cpkt, &block_ctx->ctx);
 	}
 
+	/* Handle blockwise 1 */
+	if (block_ctx) {
```

Comment:
Are we moving this up because of the removal of zoap_header_set_code()?

Reply:
Yes, there is a problem..  (thank you for bringing it up)  I need to split this block so that msg->code can be set correctly.  But  coap_append_block1_option() needs to be called after lwm2m_init_message.